doc/examples/color_exposure/plot_adapt_hist_eq_3d.py

```diff
@@ -0,0 +1,250 @@
+#############
+
+def get_rgba(scalars, cmap, vmin=0., vmax=1., alpha=0.2):
+    """
+    Convert array of scalars into array of corresponding RGBA values.
+    """
+    norm = colors.Normalize(vmin=vmin, vmax=vmax)
+    scalar_map = cm.ScalarMappable(norm=norm, cmap=cmap)
+    rgbas = scalar_map.to_rgba(scalars)
+    rgbas[:, 3] = alpha
+    return rgbas
+
+
+def plt_render_volume(vol, fig_ax, cmap,
+                      vmin=0, vmax=1,
+                      bin_widths=None, n_levels=20):
+    """
+    Render a volume in a 3D matplotlib scatter plot.
+    Better would be to use napari.
+    """
+    vol = np.clip(vol, vmin, vmax)
+
+    xs, ys, zs = np.mgrid[0:vol.shape[0]:bin_widths[0],
+                          0:vol.shape[1]:bin_widths[1],
+                          0:vol.shape[2]:bin_widths[2]]
+    vol_scaled = vol[::bin_widths[0],
+                     ::bin_widths[1],
+                     ::bin_widths[2]].flatten()
+
+    # define alpha transfer function
+    levels = np.linspace(vmin, vmax, n_levels)
+    alphas = np.linspace(0, .7, n_levels)
+    alphas = alphas ** 11
+    alphas = (alphas - alphas.min()) / (alphas.max() - alphas.min())
+    alphas *= 0.8
+
+    # group pixels by intensity and plot separately,
+    # as 3D scatter does not accept arrays of alpha values
+    for il in range(1, len(levels)):
+        sel = (vol_scaled >= levels[il - 1])
+        sel *= (vol_scaled <= levels[il])
+        if not np.max(sel):
+            continue
+        c = get_rgba(vol_scaled[sel], cmap,
+                     vmin=vmin, vmax=vmax, alpha=alphas[il - 1])
+        fig_ax.scatter(xs.flatten()[sel],
+                       ys.flatten()[sel],
+                       zs.flatten()[sel],
+                       c=c, s=0.5 * np.mean(bin_widths),
+                       marker='o', linewidth=0)
+
+
+#############
+# create figure with subplots
+#############
+
+cmap = 'Blues'
+
+fig = plt.figure(figsize=(10, 6))
+axs = [fig.add_subplot(2, 3, i + 1,
+                       projection=Axes3D.name, facecolor="none")
+       for i in range(6)]
+ims = [im_orig, im_orig_he, im_orig_ahe,
+       im_degraded, im_degraded_he, im_degraded_ahe]
+
+# prepare lines for the various boxes to be plotted
+verts = np.array([[i, j, k] for i in [0, 1]
+                  for j in [0, 1] for k in [0, 1]]).astype(np.float32)
+lines = [np.array([i, j]) for i in verts
+         for j in verts if np.allclose(np.linalg.norm(i - j), 1)]
+
+# "render" volumetric data
+for iax, ax in enumerate(axs[:]):
+    plt_render_volume(ims[iax], ax, cmap, 0, 1, [2, 2, 2], 20)
+
+    # plot 3D box
+    rect_shape = np.array(im_orig.shape) + 2
+    for line in lines:
+        ax.plot((line * rect_shape)[:, 0] - 1,
+                (line * rect_shape)[:, 1] - 1,
+                (line * rect_shape)[:, 2] - 1,
+                linewidth=1, color='grey')
+
+# add boxes illustrating the kernels
+ns = np.array(im_orig.shape) // kernel_size - 1
+for iax, ind, rect_shape in zip([1] + [2] * 4,
```

Comment:
`iax, ind` could be renamed to improve readibility.

Reply:
Renamed to `axis_ind, vertex_ind, box_shape`, hope that's clearer.